addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,874 @@
+            await this._chatbotCheckWelcomeMessages();
+        }
+
+        return superMethod.apply(this, superArguments).then(() => {
+            if (this._isChatbotRedirecting) {
+                return;
+            }
+
+            if (this._isChatbot && this._chatbotCurrentStep) {
+                if (this._chatbotCurrentStep.chatbot_step_type === 'forward_operator' &&
+                    this._chatbotCurrentStep.chatbot_operator_found) {
+                    return;  // operator has taken over the conversation, let them speak
+                } else if (this._chatbotCurrentStep.chatbot_step_type === 'free_input_multi') {
+                    this._debouncedChatbotAwaitUserInput();
+                } else if (!this._chatbotShouldEndScript()) {
+                    this._chatbotSetIsTyping();
+                    this.nextStepTimeout = setTimeout(
+                        this._chatbotTriggerNextStep.bind(this), this._chatbotMessageDelay);
+                } else {
+                    this._chatbotEndScript();
+                }
+
+                this._chatbotSaveSession();
+            }
+        });
+    },
+    /**
+     * Small override to handle chatbot welcome message(s).
+     * @private
+     */
+    _sendWelcomeMessage: function () {
+        if (this._isChatbot) {
+            let welcomeMessageDelay = this._chatbotMessageDelay;
+            if (this._chatbotBatchWelcomeMessages) {
+                welcomeMessageDelay = 0;
+            }
+            this._sendWelcomeChatbotMessage(0, welcomeMessageDelay);
+        } else {
+            this._super(...arguments);
+        }
+    },
+    /**
+     * The bot can say multiple messages in quick succession as "welcome messages".
+     * (See chatbot.script#_get_welcome_steps() for more details).
+     *
+     * It is important that those messages are sent as "welcome messages", meaning manually added
+     * within the template, without creating actual mail.messages in the mail.channel.
+     *
+     * Indeed, if the end-user never interacts with the bot, those empty mail.channels are deleted
+     * by a garbage collector mechanism.
+     *
+     * About "welcomeMessageDelay":
+     *
+     * The first time we open the chat, we want to bot to slowly input those messages in one at a
+     * time, with pauses during which the end-user sees ("The bot is typing...").
+     *
+     * However, if the user navigates within the website (meaning he has an opened mail.channel),
+     * then we input all the welcome messages at once without pauses, to avoid having that annoying
+     * slow effect on every page / refresh.
+     *
+     * @private
+     */
+    _sendWelcomeChatbotMessage: function (stepIndex, welcomeMessageDelay) {
+        const chatbotStep = this._chatbot.chatbot_welcome_steps[stepIndex];
+        this._chatbotCurrentStep = chatbotStep;
+
+        if (chatbotStep.chatbot_step_message) {
+            this._addMessage({
+                id: '_welcome_' + stepIndex,
```

Comment:
It's a bit more complicated than that since I would then have to add additional changes in the related "JS model" (WebsiteLivechatMessage).

Which would mean getters/setters and also modifying some existing livechat code (there is already a concept of welcome message that uses the same trick with the "_welcome_" id).

Did not seem worth it at the time, I would like to modify the existing code as less as possible.